### PR TITLE
App

### PR DESCRIPTION
**Pull request by sbramin**
swagger UI package, returns a handler, relies on swagger.json being in root of project, or else for temp builds in /tmp/swagger.json

### CODE REVIEW

File: cmd/swagger-ui/main.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	"log"
+	"net/http"
+
+	"github.com/utilitywarehouse/swagger-ui/swaggerui"
+)
+
+func main() {
+	s := swaggerui.Handler()
+
+	log.Panic(http.ListenAndServe(":8080", s))
```
Comment (ggirtsou) [inline cmd/swagger-ui/main.go]: I think this should be log.Fatal:
```
if err := http.ListenAndServe("8080", s); err != nil {
   log.Fatal(...)
}
```

File: assetfs/fs.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+func (f *assetFile) Stat() (os.FileInfo, error) {
+	return f.FileInfo, nil
+}
+
+type dirInfo string
+
+func (d dirInfo) Name() string       { return string(d) }
+func (d dirInfo) Size() int64        { return 0 }
+func (d dirInfo) Mode() os.FileMode  { return 0 }
+func (d dirInfo) ModTime() time.Time { return time.Now() }
```
Comment (ggirtsou) [inline assetfs/fs.go]: Why is the modification time always current time?

File: swaggerui/handler.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	data := swaggerFile()
+
+	as := &assetfs.AssetStore{
+		Names: internal.AssetNames,
+		Data:  internal.Asset,
+		Info:  internal.AssetInfo,
+	}
+	fs, err := assetfs.New(as)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create static fs: %v", err))
```
Comment (ggirtsou) [inline swaggerui/handler.go]: please don't use panic, but log.Fatal

File: assetfs/prefix.go [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package assetfs
+
+import "path"
+
+// AddPrefix joins prefix and pattern and returns a sanitized pattern.
+// e.g. prefix=/a and pattern=///b/ results in path=/a/b/
+func AddPrefix(prefix, pattern string) string {
+	if len(prefix) == 0 || prefix[0] != '/' {
```
Comment (ggirtsou) [inline assetfs/prefix.go]: the if condition can be removed and just do `prefix[0] = "/"`

File: assetfs/prefix.go [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+		prefix = "/"
+	}
+	if len(pattern) == 0 {
+		return prefix
+	}
+	if prefix != pattern && pattern[len(pattern)-1:][0] == '/' {
```
Comment (ggirtsou) [inline assetfs/prefix.go]: If you're trying to get the last character, please do: `pattern[len(pattern):]` instead as `pattern[len(pattern)-1:][0]` returns byte and it won't work.

File: cmd/swagger-ui/main.go (hunk 1 above)
Comment (sbramin) [reply cmd/swagger-ui/main.go]: I don't think one should adopt a blanket ban on Panic?  The handful of uses of Panic in this all represent catastrophic issues to the program, whats your argument against Panic?  It is cleaner than Fatal, as panic unwraps the stack before exiting.

File: assetfs/prefix.go (hunk 5 above)
Comment (sbramin) [reply assetfs/prefix.go]: It is comparing to a rune so is aok

**Pull request merged**